src/nomad_simulation_parsers/parsers/vasp/xml_parser.py

```diff
@@ -0,0 +1,77 @@
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from nomad.datamodel.datamodel import (
+        EntryArchive,
+    )
+    from structlog.stdlib import (
+        BoundLogger,
+    )
+
+from nomad.parsing.file_parser.mapping_parser import MetainfoParser, Path, XMLParser
+from nomad_simulations.schema_packages.general import Simulation
+
+
+class VasprunParser(XMLParser):
+    def mix_alpha(self, mix: float, cond: bool) -> float:
+        return mix if cond else 0
+
+    def get_eigenvalues(self, array: list) -> dict[str, Any]:
+        if array is None:
+            return {}
+        transposed = np.transpose(array)
+        return dict(eigenvalues=transposed[0], occupations=transposed[1])
+
+    def get_energy_contributions(
+        self, source: dict[str, Any], **kwargs
+    ) -> list[dict[str, Any]]:
+        return [
+            c
+            for c in source
+            if c.get(f'{self.attribute_prefix}name') not in kwargs.get('exclude', [])
+        ]
+
+    def get_data(self, source: dict[str, Any], **kwargs) -> Any:
+        if source.get(self.value_key):
+            return source[self.value_key]
+        path = kwargs.get('path')
+        if path is None:
+            return
+
+        parser = Path(path=path)
+        return parser.get_data(source)
+
+    def get_forces(self, source: dict[str, Any]) -> dict[str, Any]:
+        value = self.get_data(source, path='.varray.v')
+        if value is None:
+            return {}
+        return dict(forces=value, npoints=len(value), rank=[3])
```

Comment:
Why are we still using this older style for extracting forces? Can't we use the mapping annotation here too?

Reply:
there is a bug for the def of forces i think so i have to define the rank directly here.